spec/requests/tasks_spec.rb

```diff
@@ -0,0 +1,61 @@
+describe 'TasksController' do
+  let(:task) { FactoryGirl.create(:miq_task, :state => MiqTask::STATE_FINISHED) }
+  let(:task2) { FactoryGirl.create(:miq_task, :state => MiqTask::STATE_FINISHED) }
+
+  def expect_deleted(*args)
+    args.each do |arg|
+      expect(MiqTask.find_by(:id => arg.id)).to be_nil
+    end
+  end
+
+  it 'deletes on DELETE' do
+    api_basic_authorize resource_action_identifier(:tasks, :delete, :delete)
+
+    delete(api_task_url(nil, task.id))
+
+    expect(response).to have_http_status(:no_content) # 204
+    expect_deleted(task)
+  end
+
+  it 'deletes on POST' do
+    api_basic_authorize resource_action_identifier(:tasks, :delete)
+
+    data = {
+      :action => 'delete'
+    }
+    post(api_task_url(nil, task.id), :params => data)
+
+    expect(response).to have_http_status(:ok) # 200
+    expect_deleted(task)
+
+    expected = {
+      'success' => true,
+      'message' => "tasks id: #{task.id} deleting"
+    }
+    expect(response.parsed_body).to include(expected)
+  end
+
+  it 'bulk deletes' do
+    api_basic_authorize collection_action_identifier(:tasks, :delete)
+
+    data = {
+      :action    => 'delete',
+      :resources => [
+        {:href => api_task_url(nil, task.id)},
+        {:href => api_task_url(nil, task2.id)}
```

Comment:
Hi @himdel minor, but on all the api_task_url calls, prefer just the object being specified i.e. 

```
api_task_url(nil, task)
```